src/antsibull_core/utils/hashing.py

```diff
@@ -35,3 +36,20 @@ async def verify_hash(
         return False
 
     return True
+
+
+async def verify_a_hash(filename: str, hash_digests: Mapping[str, str]) -> bool:
+    """
+    Verify whether a file has a given hash, given a set of digests with different algorithms.
+    Will only test trustworthy hashes and return ``False`` if none matches.
+
+    :arg filename: The file to verify the hash of.
+    :arg hash_digest: A mapping of hash types to digests.
+    :returns: True if the hash matches, otherwise False.
+    """
+    for algorithm in ("sha256", "blake2b_256"):
```

Comment:
Would this tuple be better as a global constant?

Reply:
22b2d2600d5ec7ca5c0b95544c056b5cddd5054b. I also noticed that `blake2b_256` is not something you can just use, so I had to invest some more work...